rotkehlchen/globaldb/handler.py

```diff
@@ -149,63 +146,60 @@ def add_asset(
-                'identifier, type, name, symbol, started, swapped_for, '
-                'coingecko, cryptocompare, details_reference) '
-                'VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)',
-                (
-                    asset_id,
-                    asset_type.serialize_for_db(),
-                    name,
-                    symbol,
-                    started,
-                    swapped_for,
-                    coingecko,
-                    cryptocompare,
-                    details_id),
-            )
+            with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                if asset_type == AssetType.ETHEREUM_TOKEN:
+                    token = cast(EthereumToken, data)
+                    GlobalDBHandler().add_ethereum_token_data(write_cursor, token)
+                    details_id = token.ethereum_address
+                    name = token.name
+                    symbol = token.symbol
+                    started = token.started
+                    swapped_for = token.swapped_for.identifier if token.swapped_for else None
+                    coingecko = token.coingecko
+                    cryptocompare = token.cryptocompare
+                else:
+                    details_id = asset_id
+                    data = cast(Dict[str, Any], data)
+                    # The data should already be typed (as given in by marshmallow)
+                    name = data.get('name', None)
+                    symbol = data.get('symbol', None)
+                    started = data.get('started', None)
+                    swapped_for_asset = data.get('swapped_for', None)
+                    swapped_for = swapped_for_asset.identifier if swapped_for_asset else None
+                    coingecko = data.get('coingecko', None)
+                    cryptocompare = data.get('cryptocompare', '')
+
+                write_cursor.execute(
+                    'INSERT INTO assets('
+                    'identifier, type, name, symbol, started, swapped_for, '
+                    'coingecko, cryptocompare, details_reference) '
+                    'VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)',
+                    (
+                        asset_id,
+                        asset_type.serialize_for_db(),
+                        name,
+                        symbol,
+                        started,
+                        swapped_for,
+                        coingecko,
+                        cryptocompare,
+                        details_id),
+                )
+
+                # for asset details we have to add them after the addition of the main asset table
+                if asset_type != AssetType.ETHEREUM_TOKEN:
+                    asset_data = cast(Dict[str, Any], data)
+                    asset_data['identifier'] = asset_id
+                    GlobalDBHandler().add_common_asset_details(write_cursor, asset_data)
         except sqlite3.IntegrityError as e:
-            connection.rollback()
```

Comment:
In the case of an error what happens now? does it by default rollback?

Reply:
yes